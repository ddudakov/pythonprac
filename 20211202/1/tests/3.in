class C(metaclass=dump):
    def __init__(self, val, value):
        self.val = val
        self.value = value

    def add(self, other, another=None, end = 0):
        return self.val + other + (another or self.value) + end

    def sub(self, other, another=None, end=0):
        return self.val*10 - other - (another or self.value) - end

c = C(10, 5)
print(c.add(9, end = 8))
print(c.sub(9,another=10, end = 7))
