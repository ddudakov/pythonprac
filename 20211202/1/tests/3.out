__init__: (10, 5), {}
add: (9,), {'end': 8}
32
sub: (9,), {'another': 10, 'end': 7}
74
